src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -846,17 +846,24 @@ public async Task<MethodInfoWithDebugInformation> GetMethodInfo(int methodId, Ca
                 }
             }
 
+            string methodName = await GetMethodName(methodId, token);
             if (method == null)
             {
-                logger.LogDebug($"Unable to find method token: {methodToken} assembly name: {asm.Name}");
-                return null;
+                //get information from runtime
+                method = await CreateMethodInfoFromRuntimeInformation(asm, methodId, methodName, methodToken, token);
             }
 
-            string methodName = await GetMethodName(methodId, token);
             methods[methodId] = new MethodInfoWithDebugInformation(method, methodId, methodName);
             return methods[methodId];
         }
 
+        public async Task<MethodInfo> CreateMethodInfoFromRuntimeInformation (AssemblyInfo asm, int methodId, string methodName, int methodToken, CancellationToken token )
+        {
+            var typeToken = await GetTypeTokenFromMethod(methodId, token);
+            asm.TypesByToken.TryGetValue(typeToken, out TypeInfo typeInfo);
```

Comment:
When can this fail? If it will never fail then use `asm.TypesByToken[typeToken]`.
If it can, then check if it was not found, and `return null` .